lib/streamlit/cli.py

```diff
@@ -118,17 +120,74 @@ def main_hello():
     _main_run(filename)
 
 
+def _compose_option_parameter(config_option):
+    """
+    Composes given config option options as options for click lib.
+    """
+    option = "--{}".format(config_option.key)
+    param = config_option.key.replace(".", "_")
+    description = config_option.description
+    if config_option.deprecated:
+        description += "\n {} - {}".format(
+            config_option.deprecation_text, config_option.deprecation_date
+        )
+
+    return {
+        "param": param,
+        "description": description,
+        "type": config_option.type,
+        "option": option,
+    }
+
+
+def configurator_options(func):
+    """
+    Decorator that adds config param keys to click dynamically.
+    """
+    for _, value in reversed(_config._config_options.items()):
+        parsed_parameter = _compose_option_parameter(value)
+        config_option = click.option(
+            parsed_parameter["option"],
+            parsed_parameter["param"],
+            help=parsed_parameter["description"],
+            type=parsed_parameter["type"],
+        )
+        func = config_option(func)
+    return func
+
+
 @main.command("run")
+@configurator_options
 @click.argument("file_or_url", required=True)
 @click.argument("args", nargs=-1)
-def main_run(file_or_url, args=None):
+def main_run(file_or_url, args=None, **kwargs):
     """Run a Python script, piping stderr to Streamlit.
     The script can be local or it can be an url. In the
     latter case, streamlit will download the script to a
     temporary file and runs this file.
```

Comment:
This is not your doing, but can you make this docstring extend all the way to the 80-character line?